target/linux/x86/base-files/lib/upgrade/platform.sh

```diff
@@ -70,7 +112,59 @@ platform_do_bootloader_upgrade() {
 }
 
 platform_do_upgrade() {
-	local diskdev partdev diff
+	local _alt altboot bootpart defboot diff diskdev partdev partdevalt part size start
+
+	#extract the boot sector from the image
+	get_image "$@" | dd of=/tmp/image.bs count=63 bs=512b 2>/dev/null
+
+	if is_efi_system; then
+		export_partdevice_label bootpart openwrt_boot \
+			&& export_partdevice_label partdev openwrt_rootfs \
+			&& export_partdevice_label partdevalt openwrt_rootfs_alt || {
+				echo "Unable to determine upgrade device"
+				return 1
+			}
+
+		grep -q 'root=PARTLABEL=openwrt_rootfs_alt' /proc/cmdline && _alt= || _alt=_alt
+		grep -q ' /boot ' /proc/mounts || {
+			mount /dev/$bootpart /boot
```

Comment:
Why not mount it `ro`, then whenever it needs to be touched, do a:

```
mount -o remount,rw /boot
```

and then back again when you're done?

Reply:
Sure, I like that, I'll change it.

Reply:
In the end I decided to follow what openwrt does everywhere else, which is not to mount it at all in an automatic way, only on demand during upgrades.